File: src/main/java/com/vaadin/demo/component/slider/RangeSliderAccessibleName.java
package com.vaadin.demo.component.slider;

import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.slider.IntegerRangeSlider;
import com.vaadin.flow.component.slider.IntegerRangeSliderValue;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("range-slider-accessible-name")
public class RangeSliderAccessibleName extends Div {

    public RangeSliderAccessibleName() {
        // tag::snippet[]
        IntegerRangeSlider rangeSlider = new IntegerRangeSlider("Price range",
                0, 1000);
        rangeSlider.setValue(new IntegerRangeSliderValue(200, 800));
        rangeSlider.setAccessibleNameStart("Minimum price");
        rangeSlider.setAccessibleNameEnd("Maximum price");
        add(rangeSlider);
        // end::snippet[]
    }

    public static class Exporter // hidden-source-line
            extends DemoExporter<RangeSliderAccessibleName> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/slider/range-slider-accessible-name.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/slider/vaadin-range-slider.js';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('range-slider-accessible-name')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-range-slider
        label="Price range"
        min="0"
        max="1000"
        .value="${[200, 800]}"
        accessible-name-start="Minimum price"
        accessible-name-end="Maximum price"
      ></vaadin-range-slider>
      <!-- end::snippet[] -->
    `;
  }
}
